### PR TITLE
feat: Add forecasted e2e test

### LINKED ISSUE #2423: [Bug]: consumption/budget - add parameter for thresholdType
### Description

The module consumption/budget have the properties thresholdType hardcoded to 'Actual' and can't be used for forecast notifications.

### PR DESCRIPTION
**Pull request by segraef**
## Description

Closes https://github.com/Azure/bicep-registry-modules/issues/2423

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.consumption.budget](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.consumption.budget.yml)     |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+
+metadata name = 'Using `thresholdType` `Forecasted`'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
```
Comment (AlexanderSehr) [inline avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep:4]: Description needs updating (and then the readme a refresh too)

File: avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'cbmin'
```
Comment (AlexanderSehr) [inline avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep:14]: Will conflcit with defaults test

File: avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep (hunk 1 above)
Comment (segraef) [reply avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep:4]: done

File: avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep (hunk 2 above)
Comment (segraef) [reply avm/res/consumption/budget/tests/e2e/forecasted/main.test.bicep:14]: done

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**